src/tools/perf/ucp_tests.cc

```diff
@@ -327,6 +357,7 @@ class ucp_perf_test_runner {
             }
         }
 
+        wait_window(m_max_outstanding);
```

Comment:
Why this is better then `ucp_ep_flush?` And should not it be inside `UCX_PERF_TEST_FOREACH`? Outstanding requests can affect performance of the next test. Reproducibility in the bench is important...

Reply:
ucp_ep_flush() does not guarantee tag requests are completed. for example, it would not wait for rendezvous protocol.

Reply:
> should not it be inside UCX_PERF_TEST_FOREACH?

Reply:
no, because we need to have multiple outstanding requests on the fly. in the loop, we check that the oldest one has completed before starting a new one. and after the loop, we wait untill all are completed.